controllers/machinedeployment_controller.go

```diff
@@ -148,14 +148,17 @@ func (r *MachineDeploymentReconciler) reconcile(_ context.Context, cluster *clus
 
 	d.Labels[clusterv1.ClusterLabelName] = d.Spec.ClusterName
 
+	// If no selector has been provided, add label and selector for the
+	// MachineDeployment's name as a default way of providing uniqueness.
+	if len(d.Spec.Selector.MatchLabels) == 0 && len(d.Spec.Selector.MatchExpressions) == 0 {
```

Comment:
Might be good to add this in the defaulting webhook instead.

@vincepri want to get your thoughts/feedback here as well.

Reply:
I actually agree with that, I temporarily forgot those existed.